src/ring/adapter/jetty9.clj

```diff
@@ -155,9 +156,62 @@
           (.addExcludeProtocols context-server protocols))))
     context-server))
 
-(defn- https-connector [server http-configuration ssl-context-factory h2? port host max-idle-time]
+(defn- http2-server-connection-factory
+  "Set AbstractHTTP2ServerConnectionFactory specific options for connection factory presumably created from HttpConfiguration"
+  ([^AbstractHTTP2ServerConnectionFactory factory-from-http-config]
+   (http2-server-connection-factory factory-from-http-config nil))
+  ([^AbstractHTTP2ServerConnectionFactory factory-from-http-config h2-options]
+   (let [{:keys [connect-protocol-enabled
+                 ^FlowControlStrategy$Factory flow-control-strategy-factory
+                 initial-session-recv-window initial-stream-recv-window max-concurrent-streams max-dynamic-table-size
+                 max-frame-length max-header-block-fragment max-setting-keys
+                 ^RateControl$Factory rate-control-factory
+                 stream-idle-timeout use-input-direct-byte-buffers use-output-direct-byte-buffers]}
+         h2-options]
+     (cond-> factory-from-http-config
+       (boolean? connect-protocol-enabled)
+       (doto (.setConnectProtocolEnabled connect-protocol-enabled))
+
+       (contains? h2-options :flow-control-strategy-factory)
+       (doto (.setFlowControlStrategyFactory flow-control-strategy-factory))
+
+       initial-session-recv-window
+       (doto (.setInitialSessionRecvWindow initial-session-recv-window))
+
+       initial-stream-recv-window
+       (doto (.setInitialStreamRecvWindow initial-stream-recv-window))
+
+       max-concurrent-streams
+       (doto (.setMaxConcurrentStreams max-concurrent-streams))
+
+       max-dynamic-table-size
+       (doto (.setMaxDynamicTableSize max-dynamic-table-size))
+
+       max-frame-length
+       (doto (.setMaxFrameLength max-frame-length))
+
+       max-header-block-fragment
+       (doto (.setMaxHeaderBlockFragment max-header-block-fragment))
+
+       max-setting-keys
+       (doto (.setMaxSettingsKeys max-setting-keys))
+
+       rate-control-factory
+       (doto (.setRateControlFactory rate-control-factory))
+
+       stream-idle-timeout
+       (doto (.setStreamIdleTimeout stream-idle-timeout))
+
+       (boolean? use-input-direct-byte-buffers)
+       (doto (.setUseInputDirectByteBuffers use-input-direct-byte-buffers))
+
+       (boolean? use-output-direct-byte-buffers)
+       (doto (.setUseOutputDirectByteBuffers use-output-direct-byte-buffers))))))
```

Comment:
Have checked all options. All of them but `setFlowControlStrategyFactory` throws on `null`/`nil`. `setFlowControlStrategyFactory` sets the parameter on `null` for factory.

Not sure if test for it needed, because it seems like testing jetty itself.

Reply:
Thank you for fast response.

I still think we can simply use `(contains? h2-options :...)` pattern for all these items because it will work no matter how jetty's API changes.

Another issue with `boolean?` check is if user provide `:use-input-direct-byte-buffers 1`, the value will be ignored and she won't receive any feedback until finding it our by herself.

Reply:
Fair point :)

Have change to check only for the fact that option is provided by user.